### PR TITLE
Fixed Steam Roller AI check affecting Terrain setting effects

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Cleanup from #4476

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -439,9 +439,9 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
             return TRUE;
         break;
     case EFFECT_HIT_SET_REMOVE_TERRAIN:
```
Comment (AsparagusEduardo) [inline src/battle_ai_util.c:441]: Actually, this change also break AI for the "Genesis Supernova" effect of setting terrain.

File: src/battle_ai_util.c [hunk 2]
```diff
@@ -439,9 +439,9 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
...
-	if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY))
-	    return TRUE;
-	break;
+        if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY))
```
Comment (AsparagusEduardo) [inline src/battle_ai_util.c:442]:
```suggestion
        if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY) && (gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_HIT || gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_FAIL)
```

File: src/battle_ai_util.c (hunk 1 above)
Comment (AsparagusEduardo) [reply src/battle_ai_util.c:441]: Added a check for the effect's arguments.

File: src/battle_ai_util.c [hunk 3]
```diff
@@ -439,9 +439,9 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
             return TRUE;
         break;
     case EFFECT_HIT_SET_REMOVE_TERRAIN:
-	if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY))
-	    return TRUE;
-	break;
+        if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY) && (gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_HIT || gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_FAIL))
```
Comment (AlexOn1ine) [inline src/battle_ai_util.c:442]:
```suggestion
        if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY) && gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_FAIL)
```

Ice Spinner doesn't fail if there is no terrain.

Comment (AsparagusEduardo) [reply src/battle_ai_util.c:442]: Sure thing, as soon as I'm back on the computer

Comment (AsparagusEduardo) [reply src/battle_ai_util.c]: Done :)

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**